pkg/client/logs_test.go

```diff
@@ -62,7 +63,7 @@ func TestLateErrors(t *testing.T) {
 
 	// We are guaranteed to eventually get the error because we never close bytestream.
 	errcount := 0
-	for i := 0; i <= 3; i++ {
+	for i := 0; i <= 10; i++ {
```

Comment:
Was just while debugging; will see if it needs to increase at all. It may need to increase 1-2 times since there are the weird cases where 0 bytes are read because of a channel closing. I preferred that to using `goto` type statements to restart the method.

Reply:
Slightly unsure about that approach now; I tried to double check the `io.Reader` documentation and it says:
```
Implementations of Read are discouraged from returning a zero byte count with a nil error, except when len(p) == 0. Callers should treat a return of 0 and nil as indicating that nothing happened; in particular, it does not indicate EOF.

Implementations must not retain p.
```

Currently, we have 2 problems:
 - if n==0 we currently retain `p` which is a `must not` directive.
 - we return `0, nil` which is discouraged by not completely inappropriate.

Reply:
By making the method a _tad_ recursive we solve this. If in the `len(data)==0` case we call into the method again, we will either get new data or have EOF eventually.

Reply:
If you can double-check just this section of logic really; I think it is fine but would appreciate your 👀 and a manual check or two.